lib/url.c

```diff
@@ -2542,6 +2551,30 @@ static CURLcode parse_proxy(struct Curl_easy *data,
     result = CURLE_OUT_OF_MEMORY;
     goto error;
   }
+#ifdef USE_UNIX_SOCKETS
+  if(sockstype && strcmp(UNIX_SOCKET_PREFIX, host) == 0) {
```

Comment:
We usually avoid comparisons with zero like this and instead use the not operator. But the host name check should probably also be case insensitive. Try using `strcasecompare()`but note the return code.
~~~c
if(sockstype && strcasecompare(UNIX_SOCKET_PREFIX, host) ) {

Reply:
Fixed